utilities/fault_injection_fs.cc

```diff
@@ -92,28 +95,28 @@ IOStatus TestFSDirectory::Fsync(const IOOptions& options, IODebugContext* dbg) {
   if (!fs_->IsFilesystemActive()) {
     return fs_->GetError();
   }
-  {
-    IOStatus in_s = fs_->InjectMetadataWriteError();
-    if (!in_s.ok()) {
-      return in_s;
-    }
+
+  IOStatus s = fs_->MaybeInjectThreadLocalMetadataWriteError(options);
+  if (!s.ok()) {
+    return s;
   }
+
   fs_->SyncDir(dirname_);
-  IOStatus s = dir_->Fsync(options, dbg);
-  {
-    IOStatus in_s = fs_->InjectMetadataWriteError();
-    if (!in_s.ok()) {
-      return in_s;
-    }
```

Comment:
If the actual metadata write operation succeeds, we should not inject error IMO. Similar for below change.